spyder/plugins/editor/plugin.py

```diff
@@ -205,8 +205,19 @@ def __init__(self, parent, ignore_last_opened_files=False):
                 shebang = ['#!/usr/bin/env python3']
             header = shebang + [
                 '# -*- coding: utf-8 -*-',
-                '"""', 'Created on %(date)s', '',
-                '@author: %(username)s', '"""', '', '']
+                '"""',
+                '$|$',
+                '',
+                'Created on ${date}',
+                '',
+                '@author: ${username}',
+                '@file: ${file}',
+                '@project: ${projectname}',
+                '',
+                'Copyright ${year}  ${fullname}',
```

Comment:
Well, that's to everyone's liking. That's why we have templates, don't we?

Personally I always find it easier to delete tokens than to look up, which tokens are available. That's why I added maybe to many to the default?

There are three reasons why I added `${file}` and `${projectname}`: Firstly, sometimes I move around or rename files inside a project or even split a project into two. If you were to do it with out using git, it is so much easier to move them back to their original place. Secondly, files published tend to appear all over the Net and you never know, where they originally come from. This information might help (mind you, it's not foolproof, because this information can be deleted). Thirdly, I once accidentally deleted a bunch of files. Recovering them gave me their contents, but not their name or place in the directory tree.

Re the copyright notice: If someone reused your file in one of his/her projects, the copyright notice often gets lost; often not of malice, but because it is cumbersome to track down the original copyright holder and insert it into the file's header. Including the copyright notice per default makes this in my view so much easier.

What do you think? I do not have a strong feeling to remove them, but they might be useful. What do the users say? You probably know that better than I do.